src/index.ts

```diff
@@ -23,24 +24,33 @@ import * as path from 'path';
 
 const CHECK_NAME = 'Lint';
 const traverseTask = array.traverse(T.task);
-type fileWithContent = { file: string; content: string };
 
 const createSpectralAnnotations = (ruleset: string, parsed: fileWithContent[], basePath: string) =>
   pipe(
     createSpectral(ruleset),
     TE.chain(spectral => {
       const spectralRuns = parsed.map(v =>
         pipe(
-          runSpectral(spectral, v.content),
-          TE.map(rules => ({ path: v.file, rules }))
+          runSpectral(spectral, v),
+          TE.map(results => {
+            info(`Done linting '${v.path}'`);
+
```

Comment:
Is it really worth to add logging here? It's a side effect and it should be encapsulated in an `IO` .

It would better to use a Writer or simply hold off on logging here